provider/resource_collection.go

```diff
@@ -501,7 +501,7 @@ func collectionApiEntriesDeleteOldRows(ctx context.Context, d *schema.ResourceDa
 
 	gen := api.Data["generation"].(int)
 	scope := api.Data["scope"].(string)
-	q := fmt.Sprintf("{\"$or\":[{\"_gen\":null},{\"_gen\":{\"$lt\":\"%d\"}}]}", gen)
+	q := fmt.Sprintf("{\"$or\":[{\"_gen\":null},{\"_gen\":{\"$ne\": %d}}]}", gen)
```

Comment:
nice

Reply:
If we end up with _gen=9 & _gen=12. what is guarantee us that 9 ones will be deleted?

Reply:
this will remove all but the current _gen. If current _gen in 12, it will delete everything else

Reply:
thank you